ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java

```diff
@@ -683,7 +716,154 @@ public void handleReplicaMetadataRequest(NetworkRequest request) throws IOExcept
 
   @Override
   public void handleReplicateBlobRequest(NetworkRequest request) throws IOException, InterruptedException {
-    throw new UnsupportedOperationException("Request type not supported");
+    if (connectionPool == null || transformer == null) {
```

Comment:
Minor: May be we don't need to check if `transformer == null` since `MessageSievingInputStream.transferInputStream(transformer)` returns original message if transformer is null?

Reply:
Actually I have a question here.
If the transfer is null for some reason, shall we keep the original message and write to the store or fail the request?
For example, we may fail to get the transformer below. 
I'm not exactly sure what the transfer does. If we are supposed to do the transfer but fails to do that, does that mean the content is not we want and we shouldn't write to the store?


      try {
        StoreKeyConverter keyConverter = storeKeyConverterFactory.getStoreKeyConverter();
        transformer = Utils.getObj(serverConfig.serverMessageTransformer, storeKeyFactory, keyConverter);
      } catch (Exception e) {
        logger.error("Failed to create transformer", e);
      }

Reply:
Hmm, just thinking that since functionality of `transformer` is abstracted out, we could carry on in this method even if it is null. But, may be either way is okay.